google/cloud/universe_domain/demo/README.md

```diff
@@ -0,0 +1,157 @@
+`GOOGLE_APPLICATION_CREDENTIALS` environment variable to find the credentials
+file. For example:
+
+| Shell              | Command                                                                              |
+| :----------------- | ------------------------------------------------------------------------------------ |
+| Bash/zsh/ksh/etc.  | `export GOOGLE_APPLICATION_CREDENTIALS=[PATH]`                                       |
+| sh                 | `GOOGLE_APPLICATION_CREDENTIALS=[PATH];`<br> `export GOOGLE_APPLICATION_CREDENTIALS` |
+| csh/tsch           | `setenv GOOGLE_APPLICATION_CREDENTIALS [PATH]`                                       |
+| Windows Powershell | `$env:GOOGLE_APPLICATION_CREDENTIALS=[PATH]`                                         |
+| Windows cmd.exe    | `set GOOGLE_APPLICATION_CREDENTIALS=[PATH]`                                          |
+
+Setting this environment variable is the recommended way to configure the
+authentication preferences, though if the environment variable is not set, the
+library searches for a credentials file in the same location as the
+[Cloud SDK](https://cloud.google.com/sdk/). For more information about
+*Application Default Credentials*, see
+https://cloud.google.com/docs/authentication/production
+
+If the file found at `GOOGLE_APPLICATION_CREDENTIALS` contains the
+`universe_domain` field, that value is used to determine service endpoints. If
+the `universe_domain` field is not present, the default value of
+`googleapis.com` is used.
+
+## Using with Bazel
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+1. Install Bazel using [the instructions][bazel-install] from the `bazel.build`
+   website.
+
+1. Compile this example using Bazel:
+
+   ```bash
+   cd $HOME/google-cloud-cpp/google/cloud/universe_domain/demo
+   bazel build ...
+   ```
+
+   Note that Bazel automatically downloads and compiles all dependencies of the
+   project. As it is often the case with C++ libraries, compiling these
+   dependencies may take several minutes.
+
+1. Run the example, changing the placeholder(s) to appropriate values:
+
+   ```bash
+   bazel run :kms_demo -- [PROJECT_ID] [LOCATION_ID]
+   ```
+
+## Using with CMake
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+1. Install CMake. The package managers for most Linux distributions include a
+   package for CMake. Likewise, you can install CMake on Windows using a package
+   manager such as [chocolatey][choco-cmake-link], and on macOS using
+   [homebrew][homebrew-cmake-link]. You can also obtain the software directly
+   from the [cmake.org](https://cmake.org/download/).
+
+1. Install the dependencies with your favorite tools. As an example, if you use
+   [vcpkg](https://github.com/Microsoft/vcpkg.git):
+
+   ```bash
+   cd $HOME/vcpkg
+   ./vcpkg install google-cloud-cpp[core,kms,universe_domain]
```

Comment:
comment: This looks like a chicken and egg thing. AFAIK our vcpkg port does not have a universe_domain feature.

I think we have this problem for any new library before it is released, though.